src/libs/actions/BankAccounts.js

```diff
@@ -114,9 +115,170 @@ function addPlaidBankAccount(account, password, plaidLinkToken) {
         });
 }
 
+/**
+ * Fetch and save locally the Onfido SDK token and applicantID
+ * - The sdkToken is used to initialize the Onfido SDK client
+ * - The applicantID is combined with the data returned from the Onfido SDK as we need both to create an
+ *   identity check. Note: This happens in Web-Secure when we call Activate_Wallet during the OnfidoStep.
+ */
+function fetchOnfidoToken() {
+    // Use Onyx.set() since we are resetting the Onfido flow completely.
+    Onyx.set(ONYXKEYS.WALLET_ONFIDO, {loading: true});
+    API.Wallet_GetOnfidoSDKToken()
+        .then((response) => {
+            const apiResult = lodashGet(response, ['requestorIdentityOnfido', 'apiResult'], {});
+            Onyx.merge(ONYXKEYS.WALLET_ONFIDO, {
+                applicantID: apiResult.applicantID,
+                sdkToken: apiResult.sdkToken,
+                loading: false,
+                hasAcceptedPrivacyPolicy: true,
+            });
+        });
+}
+
+/**
+ * Privately used to update the additionalDetails object in Onyx (which will have various effects on the UI)
+ *
+ * @param {Boolean} loading whether we are making the API call to validate the user's provided personal details
+ * @param {String[]} [errorFields] an array of field names that should display errors in the UI
+ * @param {String} [additionalErrorMessage] an additional error message to display in the UI
+ * @private
+ */
+function setAdditionalDetailsStep(loading, errorFields = null, additionalErrorMessage = '') {
+    Onyx.merge(ONYXKEYS.WALLET_ADDITIONAL_DETAILS, {loading, errorFields, additionalErrorMessage});
+}
+
+/**
+ * This action can be called repeatedly with different steps until an Expensify Wallet has been activated.
+ *
+ * Possible steps:
+ *
+ *     - OnfidoStep - Creates an identity check by calling Onfido's API (via Web-Secure) with data returned from the SDK
+ *     - AdditionalDetailsStep - Validates a user's provided details against a series of checks
+ *     - TermsStep - Ensures that a user has agreed to all of the terms and conditions
+ *
+ * The API will always return the updated userWallet in the response as a convenience so we can avoid calling
+ * Get&returnValueList=userWallet after we call Wallet_Activate.
+ *
+ * @param {String} currentStep
+ * @param {Object} parameters
+ * @param {String} [parameters.onfidoData] - JSON string
+ * @param {Object} [parameters.personalDetails] - JSON string
+ * @param {Boolean} [parameters.hasAcceptedTerms]
+ */
+function activateWallet(currentStep, parameters) {
+    let personalDetails;
+    let onfidoData;
+    let hasAcceptedTerms;
```

Comment:
Should these be defined at the top of the module? As it is, these will be reset every time the method is called, but I don't think that's the intention.

Reply:
>these will be reset every time the method is called, but I don't think that's the intention

That actually was my intention and so, no they shouldn't be defined at the top of the module. If they were then we'd keep sending the params to the API and the expected behavior is that they'd be `undefined` (at least that was my thought process when I last left this off) - not sure if having them sent again with every request will cause issues (but is definitely not required).

Reply:
Ah, OK. Thanks for explaining! I think it should be fine as long as that was the intent.